js/coinflex.js

```diff
@@ -0,0 +1,1920 @@
+            let isDepositEnabled = true;
+            const fees = {};
+            const networks = {};
+            const networkList = this.safeValue (entry, 'networkList', []);
+            for (let j = 0; j < networkList.length; j++) {
+                const networkItem = networkList[j];
+                const networkId = this.safeString (networkItem, 'network');
+                // const name = this.safeString (networkItem, 'name');
+                const depositEnable = this.safeValue (networkItem, 'canDeposit');
+                const withdrawEnable = this.safeValue (networkItem, 'canWithdraw');
+                isDepositEnabled = isDepositEnabled || depositEnable;
+                isWithdrawEnabled = isWithdrawEnabled || withdrawEnable;
+                fees[networkId] = undefined;
+                networks[networkId] = {
+                    'id': networkId,
+                    'network': networkId,
+                    'active': isDepositEnabled && isWithdrawEnabled,
+                    'deposit': isDepositEnabled,
+                    'withdraw': isWithdrawEnabled,
+                    'fee': undefined,
+                    'precision': this.safeInteger (networkItem, 'transactionPrecision'),
+                    'limits': {
+                        'deposit': {
+                            'min': this.safeNumber (networkItem, 'minDeposit'),
+                            'max': undefined,
+                        },
+                        'withdraw': {
+                            'min': this.safeNumber (networkItem, 'minWithdrawal'),
+                            'max': undefined,
+                        },
+                    },
+                    'info': networkItem,
+                };
+            }
+            result[code] = {
+                'id': id,
+                'name': code,
+                'code': code,
+                'precision': undefined,
+                'info': entry,
+                'active': isWithdrawEnabled && isDepositEnabled,
+                'deposit': isDepositEnabled,
+                'withdraw': isWithdrawEnabled,
+                'fee': undefined,
+                'fees': undefined,
+                'limits': {
+                    'amount': {
+                        'min': undefined,
+                        'max': undefined,
+                    },
+                },
+                'networks': networks,
+            };
+        }
+        return result;
+    }
+
+    setStartEndTimes (request, since) {
+        // exchange has 7 days maximum allowed distance between start/end times across its api endpoints
+        const distance = 7 * 24 * 60 * 60 * 1000; // 7 days
+        if (since === undefined) {
+            since = this.milliseconds () - distance + this.timeout; // don't directly set 7 days ago from this moment, as when request arrives at exchange, it will be more than 7 days from 'current time'. so, add timeout seconds to make sure we have enough time to reach exchange.
```

Comment:
1) if that's case, so instead of `sum` we will need the opposite (which method do we have in base for opposite of sum?) 
2) however, what is the reason of arithmetical operators being transpiled as a string?

Reply:
1) Subtraction (-) is not problematic because is not an ambiguous operation, meaning it is not used elsewhere besides numbers arithmetic whereas '+' might be used to concatenate strings or sum numbers.

2) Because our transpiler is regex-based so we can't infer types at transpile time. Even with an "AST based" compiler might be impossible to correctly guess the type because JS is dynamically typed so only at runtime we can be sure of the variable's type.

Reply:
https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#adding-numbers-and-concatenating-strings